src/wp-includes/class-wp-theme.php

```diff
@@ -1706,6 +1706,73 @@ public static function get_allowed_on_site( $blog_id = null ) {
 		return (array) apply_filters( 'site_allowed_themes', $allowed_themes[ $blog_id ], $blog_id );
 	}
 
+	/**
+	 * Enables a theme for a site.
+	 *
+	 * @param string|string[] $stylesheets Stylesheet name or array of stylesheet names.
+	 * @param int             $site_id     Site ID.
+	 *
+	 * @since 6.3
+	 *
+	 */
+	public static function site_enable_theme( $stylesheets, $site_id = null ) {
+		$site_id = (int) $site_id;
+
+		if ( ! is_multisite() ) {
+			return;
+		}
+
+		if ( ! is_array( $stylesheets ) ) {
+			$stylesheets = array( $stylesheets );
+		}
+
+		$current_site_id = get_current_blog_id();
```

Comment:
This line should probably inside the if statement next line, to only get called when site id is not given.